pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -291,3 +291,132 @@ func NewNamedMapFromBuildPlan(build *model.Build) (artifact.NamedMap, error) {
+	}
+	for _, step := range build.Steps {
+		lookup[step.StepID] = step
+	}
+	for _, source := range build.Sources {
+		lookup[source.NodeID] = source
+	}
+
+	for _, a := range build.Artifacts {
+		// Since we are using the BuildLogStreamer, we need to add all of the
+		// artifacts that have been submitted to be built except for the artifacts
+		// of the builder mime type as the BuildLogStreamer does not send events
+		// for builder artifacts.
+		if a.MimeType == model.XActiveStateBuilderMimeType {
+			continue
+		}
+
+		_, ok := result[strfmt.UUID(a.NodeID)]
+		if !ok && a.Status != model.ArtifactNotSubmitted {
+			deps := make(map[strfmt.UUID]struct{})
+			for _, depID := range a.RuntimeDependencies {
+				artifact, ok := lookup[depID].(*model.Artifact)
+				if ok {
+					if artifact.MimeType == model.XActiveStateBuilderMimeType {
+						logging.Debug("Skipping active state builder artifact: %s", depID)
+						continue
+					}
+				}
+
+				deps[depID] = struct{}{}
+				recursiveDeps, err := generateBuildtimeDependencies(depID, lookup, deps)
+				if err != nil {
+					return nil, errs.Wrap(err, "Could not resolve runtime dependencies for artifact: %s", depID)
+				}
+				for id := range recursiveDeps {
+					deps[id] = struct{}{}
+				}
+			}
+
+			var uniqueDeps []strfmt.UUID
+			for id := range deps {
+				if _, ok := deps[id]; !ok {
+					continue
+				}
+				uniqueDeps = append(uniqueDeps, id)
+			}
+
+			info, err := getSourceInfo(a.GeneratedBy, lookup)
+			if err != nil {
+				return nil, errs.Wrap(err, "Could not resolve source information")
+			}
+
+			result[strfmt.UUID(a.NodeID)] = artifact.Artifact{
+				ArtifactID:       strfmt.UUID(a.NodeID),
+				Name:             info.Name,
+				Namespace:        info.Namespace,
+				Version:          &info.Version,
+				RequestedByOrder: true,
+				GeneratedBy:      a.GeneratedBy,
+				Dependencies:     uniqueDeps,
+			}
+		}
+	}
+
+	return result, nil
+}
+
+func generateBuildtimeDependencies(depdendencyID strfmt.UUID, lookup map[strfmt.UUID]interface{}, result map[strfmt.UUID]struct{}) (map[strfmt.UUID]struct{}, error) {
+	artifact, ok := lookup[depdendencyID].(*model.Artifact)
+	if !ok {
+		_, sourceOK := lookup[depdendencyID].(*model.Source)
+		if sourceOK {
+			// Dependency is a source, skipping
+			return nil, nil
+		}
+		return nil, errs.New("Incorrect target type for id %s, expected Artifact or Source", depdendencyID)
+	}
+
+	if artifact.MimeType == model.XActiveStateBuilderMimeType {
+		logging.Debug("Skipping builder artifact %s", artifact.NodeID)
+		return nil, nil
+	}
+
+	for _, depID := range artifact.RuntimeDependencies {
+		result[depID] = struct{}{}
+		_, err := generateBuildtimeDependencies(depID, lookup, result)
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not build map for runtime dependencies of artifact %s", artifact.NodeID)
+		}
+	}
+
+	step, ok := lookup[artifact.GeneratedBy].(*model.Step)
+	if !ok {
+		_, ok := lookup[artifact.GeneratedBy].(*model.Source)
```

Comment:
In this scenario if `ok==true` then `step` will be nil and line 409 would panic.

Reply:
If `ok` is true we don't make it to line 409, we return `nil` before that.